src/streaming/FetchLoader.js

```diff
@@ -0,0 +1,181 @@
+ *
+ * Copyright (c) 2013, Dash Industry Forum.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *  * Redistributions of source code must retain the above copyright notice, this
+ *  list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation and/or
+ *  other materials provided with the distribution.
+ *  * Neither the name of Dash Industry Forum nor the names of its
+ *  contributors may be used to endorse or promote products derived from this software
+ *  without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS AS IS AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ *  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ *  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ *  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ *  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import FactoryMaker from '../core/FactoryMaker';
+import ISOBoxer from 'codem-isoboxer';
+
+/**
+* @module FetchLoader
+* @description Manages download of resources via HTTP using fetch.
+* @param {Object} cfg - dependancies from parent
+*/
+function FetchLoader(cfg) {
+
+    cfg = cfg || {};
+
+    let instance;
+
+    function send(httpRequest) {
+
+        // Variables will be used in the callback functions
+        let firstProgress = true; /*jshint ignore:line*/
+        let needFailureReport = true; /*jshint ignore:line*/
+        let requestStartTime = new Date();
+        let lastTraceTime = requestStartTime; /*jshint ignore:line*/
+        let lastTraceReceivedCount = 0; /*jshint ignore:line*/
+
+        let request = httpRequest.request;
+
+        const headers = new Headers(); /*jshint ignore:line*/
+        if (request.range) {
+            headers.append('Range', 'bytes=' + request.range);
+        }
+
+        if (!request.requestStartDate) {
+            request.requestStartDate = requestStartTime;
+        }
+
+        const reqOptions = {
+            method: httpRequest.method,
+            headers: headers,
+            credentials: httpRequest.withCredentials ? 'include' : undefined
+        };
+
+        fetch(httpRequest.url, reqOptions).then(function (response) {
+            if (!response.ok) {
+                httpRequest.onend();
```

Comment:
`onend` expects to read `statusText` from the response, but that doesn't appear to be set until below.